actionpack/lib/action_controller/test_case.rb

```diff
@@ -593,7 +607,14 @@ def process(action, http_method = 'GET', *args)
           @request.env['RAW_POST_DATA'] = args.shift
         end
 
-        parameters, session, flash = args
+        if args[0] && args[0].keys.any? { |k| REQUEST_KWARGS.include?(k) }
+          parameters = args[0][:params]
+          session =    args[0][:session]
+          flash =      args[0][:flash]
+        else
+          parameters, session, flash = args
```

Comment:
New form should be:

`process action, method: 'GET', params: {}, session: {}, flash: {}`

And then we add on with:

`post/get/put/patch/delete/head action, params: {}, session: {}, flash: {}`